typescript/common-resolvers/src/dataset.ts

```diff
@@ -161,6 +163,124 @@ const createDataset = async (
   }
 };
 
+export const createTutorialDataset = async (
+  _: any,
+  args: MutationCreateTutorialDatasetArgs,
+  { repository, req, user }: Context
+): Promise<any> => {
+  const now = new Date().toISOString();
+  const datasetId = args.data.id ?? uuidv4();
+  const datasetName = trim(args.data.name);
+
+  if (datasetName === "") {
+    throw new Error("Could not create the dataset with an empty name");
+  }
+
+  const dbDataset = {
+    id: datasetId,
+    createdAt: now,
+    updatedAt: now,
+    name: datasetName,
+    workspaceSlug: args.data.workspaceSlug,
+  };
+
+  try {
+    await repository.dataset.add(dbDataset, user);
+
+    const createdDataset = await getDataset(
+      { id: datasetId },
+      repository,
+      user
+    );
+    const labelClass = args?.data.labelClass;
+    const tutorialLabelClassId = uuidv4();
+    const dbLabelClass = {
+      id: tutorialLabelClassId,
+      index: 0,
+      createdAt: now,
+      updatedAt: now,
+      name: labelClass?.name ?? "",
+      color: labelClass?.color,
+      datasetId: createdDataset.id,
+    };
+    const createdLabelClass = await repository.labelClass.add(
+      dbLabelClass,
+      user
+    );
+    const datasetImages = args.data.images;
+    const workspaceId = await getWorkspaceIdOfDataset({
+      repository,
+      datasetId,
+      user,
+    });
+
+    const imagesToCreate: DbImageCreateInput[] = await Promise.all(
+      datasetImages.map((imageToImport) =>
+        importAndProcessImage(
+          { image: { ...imageToImport, datasetId }, workspaceId },
+          { repository, req }
+        )
+      )
+    );
+
+    const imageIds = await repository.image.addMany(
+      { datasetId, images: imagesToCreate },
+      user
+    );
+
+    const createdImages = await repository.image.list({
+      id: { in: imageIds },
+      user,
+    });
+
+    const imageWithLabelsId = createdImages.find(
+      (image) => image.name === "image-4"
+    );
+
+    const originalLabels = args?.data.labels;
+
+    const labelsToCreate = originalLabels.map((originalLabel) => {
+      const labelId = uuidv4();
+
+      return {
+        ...originalLabel,
+        labelClassId:
+          originalLabel.labelClassId === null ? null : tutorialLabelClassId,
+        type: originalLabel.type ?? LabelType.Polygon,
+        id: labelId,
+        createdAt: now,
+        updatedAt: now,
+        imageId: imageWithLabelsId?.id ?? "",
+      };
+    });
+
+    const createdLabels = await repository.label.addMany(
+      {
+        imageId: imageWithLabelsId?.id ?? "",
```

Comment:
Why set both `imageId` here and on each label ? (see my other comment on `label.addMany` on a related subject)

Reply:
Fixed